### DISCUSSION TITLE
Scrolling on a trackpad occasionally shows the volume dials when it shouldent

### DISCUSSION THREAD

**Discussion by 2minwia** [ppy/osu#18303] [Q&A]
For reasons-I-can't-understand, scrolling around on a trackpad (in actual menus, not empty space where it should appear) a lot sometimes shows the volume dial. It's not really a big deal, but it can be a little distracting.
It happens in most menus, have it be osu!direct (see video), or the song selection screen (while not hovering the center of the screen, that actually should give the volume dial, i.e the song wheel/ranking panel).

Video: https://streamable.com/utpw5d

System: MacOS 12.3, M1 MacBook Air (using the laptops trackpad)
Version: 2022.515.0 (apple silicon build)

[database.log](https://github.com/ppy/osu/files/8699596/database.log)
[network.log](https://github.com/ppy/osu/files/8699597/network.log)
[performance.log](https://github.com/ppy/osu/files/8699598/performance.log)
[runtime.log](https://github.com/ppy/osu/files/8699599/runtime.log)

**Comment by peppy** [comment]
You're probably scrolling using two fingers occasionally?

**Reply by smoogipoo** [reply]
Isn't that how scrolling works on a trackpad...?

**Reply by peppy** [reply]
Yes, I was explaining to the user. Although watching the video makes it look like there's something more at play (I don't generally watch `streamable` links as they contain insane ads).

**Reply by 2minwia** [reply]
Oh yeah, to clarify, the video only shows me doing two finger scrolling, no click-and-drag-to-scroll.

**Comment by frenzibyte** [comment]
Can also reproduce this, will attempt to investigate this one.

**Reply by frenzibyte** [reply]
Apparently, this is a regression from https://github.com/ppy/osu-framework/pull/5118 due to early-returning when the X delta is greater than the Y delta (which can be the case if you scroll a bit to the side as you go down). 

The `KeyBindingContainer` still picks up that scroll event and thinks it's a vertical scroll due to  [this logic](https://github.com/ppy/osu-framework/blob/9fa025c59363378fa0b37ed02641c0b300a294be/osu.Framework/Input/Bindings/KeyCombination.cs#L617-L630) which doesn't match [what `ScrollContainer` uses](https://github.com/ppy/osu-framework/blob/0fdff74e4366536aa060c2a0cc3c073a12092ec3/osu.Framework/Graphics/Containers/ScrollContainer.cs#L371).

Unsure about a specific solution, but I feel we want the key determining logic to match `ScrollContainer` rather than pressing separate scroll keys depending on delta, potentially causing chaos when one action is bound to vertical while another is bound to horizontal (cc @peppy):
```csharp
if (e.ScrollDelta.Y > e.ScrollDelta.X)
{
    if (e.ScrollDelta.Y > 0)
        return InputKey.MouseWheelUp;
    else if (e.ScrollDelta.Y < 0)
        return InputKey.MouseWheelDown;
}
else
{
    if (e.ScrollDelta.X > 0)
        return InputKey.MouseWheelLeft;
    else if (e.ScrollDelta.X < 0)
        return InputKey.MouseWheelRight;
}
```
